api/mapping/models.py

```diff
@@ -190,47 +142,44 @@ class ScanReport(BaseModel):
     """
 
     data_partner = models.ForeignKey(
-        DataPartner,
-        on_delete=models.CASCADE,
-        blank=True,
-        null=True,
+        DataPartner, on_delete=models.CASCADE, blank=True, null=True,
     )
 
     author = models.ForeignKey(
-        settings.AUTH_USER_MODEL,
-        on_delete=models.CASCADE,
-        blank=True,
-        null=True,
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, blank=True, null=True,
     )
 
-    name = models.CharField(
-        max_length=256,
-    )
+    name = models.CharField(max_length=256,)
 
-    dataset = models.CharField(
-        max_length=128,
-    )
+    # TODO: rename to `dataset_name`
+    dataset = models.CharField(max_length=128,)
 
-    hidden =  models.BooleanField(
-        default=False,
-    )
+    hidden = models.BooleanField(default=False,)
 
     file = models.FileField()
 
-    status=models.CharField(
-        max_length=7,
-        choices=Status.choices,
-        default=Status.UPLOAD_IN_PROGRESS,        
+    status = models.CharField(
+        max_length=7, choices=Status.choices, default=Status.UPLOAD_IN_PROGRESS,
     )
 
     data_dictionary = models.ForeignKey(
         "DataDictionary",
         on_delete=models.CASCADE,
         null=True,
         blank=True,
-        related_name = 'data_dictionary'
+        related_name="data_dictionary",
+    )
+
+    # TODO: rename to `dataset`
+    parent_dataset = models.ForeignKey(
+        "Dataset",
+        on_delete=models.CASCADE,
+        related_name="scan_reports",
+        related_query_name="scan_report",
+        null=True,
+        blank=True,
     )
-    
+
```

Comment:
Review here @spco. I put the note to rename it in future to a less confusing name as well as changing the one that is already called `dataset` since that one is really just the name and `name` is for something else.